server/api/src/rs_genquery2.cpp

```diff
@@ -74,7 +74,7 @@ auto rs_genquery2(RsComm* _comm, Genquery2Input* _input, char** _output) -> int
     }
 
     if (host_info->localFlag != LOCAL_HOST) {
-        log_api::trace("{}: Redirecting request to remote zone [{}].", __func__, _input->zone);
+        log_api::trace("{}: Redirecting request to server [{}].", __func__, host_info->hostName->name);
```

Comment:
why just the server name and not also the zone name?

Reply:
The zone name is an input argument from the client and is allowed to be null.

Reply:
I misunderstood your comment. The zone name could be included. I just didn't think to do that since that felt less important than knowing to which server the request was sent.

Reply:
but when it's NOT null... it IS going to a different zone, so that feels relevant and helpful.     i think that was the initial intent of the line.

Reply:
Correct. The target zone is still logged.

The original log message was slightly incorrect because it didn't cover redirects within the same zone (i.e. consumer -> provider).

Reply:
@alanking is referring to these lines. They are the first lines logged whenever the API is invoked (w/ a log level of trace).

https://github.com/irods/irods/blob/a9647277f3010ff5fe9123435bca68181d31b274/server/api/src/rs_genquery2.cpp#L40-L50

Reply:
gotcha.  resolving.